magda-typescript-common/src/registry/AuthorizedRegistryClient.ts

```diff
@@ -13,28 +13,38 @@ import createServiceError from "../createServiceError";
 import buildJwt from "../session/buildJwt";
 import { IncomingMessage } from "http";
 import { Maybe } from "tsmonad";
+import { MAGDA_ADMIN_PORTAL_ID } from "./TenantConsts";
 
 export interface AuthorizedRegistryOptions extends RegistryOptions {
     jwtSecret: string;
     userId: string;
+    tenantId: number;
```

Comment:
This should be `tenantId?` if it can be `undefined`

Reply:
The tenantId is required because access to registry is tenant dependent.

Reply:
On second thought, it would be better to make `tenantId` compulsory in both RegistryClient and AuthorizedRegistryClient to avoid confusion and error.

Reply:
The argument `tenantId` is now compulsory, which also uncovers some tenant related problems yet to be addressed in multi-tenant mode. They are marked under `TODO`.

Reply:
Sounds good :)